src/srv.rs

```diff
@@ -65,33 +75,19 @@ impl SrvResolver {
         let lookup_hostname = format!("_mongodb._tcp.{}", original_hostname);
 
         let srv_lookup = self.resolver.srv_lookup(lookup_hostname.as_str()).await?;
+        let mut srv_addresses: Vec<Result<ServerAddress>> = Vec::new();
+        let mut min_ttl = u32::MAX;
```

Comment:
the changes here are kind of hard to read, but basically it combines the `min_ttl` map and the `srv_addresses` map into a single for loop over the same information.